metroae_config.py

```diff
@@ -598,7 +607,7 @@ def read_and_parse_excel(self, excel_file):
                                       error["message"]))
             exit(1)
 
-        for template_name, var_set_entries in data.iteritems():
+        for template_name, var_set_entries in self.iteritems(data):
```

Comment:
Does this work for both python 2 and 3?

Reply:
yes, I added the following on lines 302-310 to handle the Python 2 and Python 3 case. 

`    try:
        dict.iteritems
    except AttributeError:
        # Python 3
        def iteritems(self, d):
            return iter(d.items())
    else:
        # Python 2
        def iteritems(self, d):
            return d.iteritems()`

Reply:
that didn't format the way I wanted it to. The `except` code block handles the Python 3 case and the `else` code block handles the Python 2 case

Reply:
why don't we just use items() in both cases. it is supported in both python2 and python3 Correct?

Reply:
yes, items() is supported in both cases. I didn't go that route initially because we used iteritems() not items(). The issue with using items() for python 2 is that it generates a list of all the key value pairs of the dictionary and we would then iterate over that - this could be memory intensive for large dictionaries. I think that could be a concern here, right? Iteritems() is generally preferred over items(), especially for larger dictionaries (though I'm not sure what size dictionary would really lead to noticeable differences) since it creates an iterator instead of a list

Reply:
I see your point. but in this case our data set is small enough not to cause problem and the other thing is python3 does the same thing as python2 when you call items. I would choose code simplicity in this case and not have to deal with two different method calls based on version.

Reply:
ok, sounds good. I think there is one difference between python 2 items() and python 3 items(), though. In Python 3, items() returns a view object, which means that if the dictionary is updated, the view object would also be updated. The Python 2 equivalent would be viewitems(). Items() in Python 2 just returns a copy so if the dictionary is updated, items() wouldn't be. In our case it doesn't matter since we're not changing the dictionary